CMakeLists.txt

```diff
@@ -37,19 +37,27 @@ endif()
 #-------------------------------------------------------------------------------
 # Options and settings
 #-------------------------------------------------------------------------------
+option(STABLEHLO_BUILD_EMBEDDED "Build StableHLO as part of another project" OFF)
```

Comment:
There are quite a few related boolean variables - STABLEHLO_BUILD_EMBEDDED, STABLEHLO_STANDALONE_BUILD and STABLEHLO_EXTERNAL_PROJECT_BUILD - and I think that not all combinations of them are valid, so it's a bit confusing. 

However, given that this PR aims to exactly reproduce the MLIR-HLO setup, I think we should keep these variables as is to first establish the baseline and then think about refining it. I'll maybe look into this in a follow-up PR.